ariadne/cardano/src/Ariadne/Wallet/Cardano/Kernel/Keystore/Util.hs

```diff
@@ -0,0 +1,97 @@
+-- | system file operations related to key-store UserSecret
+
+module Ariadne.Wallet.Cardano.Kernel.Keystore.Util
+       ( isEmptyUserSecret
+       , getTempKeystorePath
+       , writeUserSecret
+       , peekUserSecret
+       , readUserSecret
+       ) where
+
+import Control.Exception.Safe (finally)
+import Data.Aeson (eitherDecode, encode)
+import Data.Default (Default (..))
+import System.Directory (doesFileExist, renameFile, createDirectoryIfMissing, getTemporaryDirectory)
+import System.FilePath ((</>), takeDirectory, takeFileName)
+import System.FileLock (SharedExclusive (..), withFileLock)
+import System.IO (hClose, openBinaryTempFile)
+
+import qualified Data.ByteString.Lazy as LBS (hPut, readFile)
+
+import Ariadne.System.FileMode (ensureModeIs, mode600)
+import Ariadne.System.Other (whenPosix)
+import Ariadne.Wallet.Cardano.Kernel.Keystore.Types
+  (UserSecret(..), StorageDecodingError(..), usWallets, usFilePath)
+
+
+lockFilePath :: FilePath -> FilePath
+lockFilePath = (<> ".lock")
+
+-- | Creates empty temporary storage file for keyStore testing purposes
+getTempKeystorePath :: MonadIO m => m FilePath
+getTempKeystorePath = do
```

Comment:
It is perhaps better to wrap the whole function in `liftIO $ do`.

Reply:
I agree. 👍